src/signals.jl

```diff
@@ -129,6 +129,53 @@ Invoke/return `Legolas.write(path_or_io, signals, Schema("onda.signal@1"); kwarg
 """
 write_signals(path_or_io, signals; kwargs...) = Legolas.write(path_or_io, signals, Legolas.Schema("onda.signal@1"); kwargs...)
 
+"""
+    validate_signals(signals)
+
+Perform both table-level and row-level validation checks on the content of `signals`,
+a presumed `onda.signal` table. Returns `signals`.
+
+This function will throw an error in any of the following cases:
+
+- `Legolas.validate(signals, Legolas.Schema("onda.signal@1"))` throws an error
+- `Signal(row)` errors for any `row in Tables.rows(signals)`
```

Comment:
I somehow thought that the schema validation and the `Signal` constructor would be equivalent

Reply:
Signal constructor validates the actual values passed into it, `Legolas.validate(signals, Schema("onda.signal@1"))` validates that `Tables.schema(signal)` complies with the given schema